### PR TITLE
Parse 'customerSheetClientSecret' for Android and iOS

### PR DESCRIPTION
**Pull request by tjclawson-stripe**
## Summary

Add logic to parse `customerSessionClientSecret`

## Motivation

Support Customer Session for React Native
[MOBILESDK-2432](https://jira.corp.stripe.com/browse/MOBILESDK-2432)
[MOBILESDK-2431](https://jira.corp.stripe.com/browse/MOBILESDK-2431)

## Testing

- [X] I tested this manually
- [ ] I added automated tests

## Documentation

Select one: 
- [ ] I have added relevant documentation for my changes.
- [X] This PR does not result in any developer-facing changes.

### CODE REVIEW

File: android/src/main/java/com/reactnativestripesdk/PaymentSheetFragment.kt [hunk 1]
```diff
@@ -191,7 +193,11 @@ class PaymentSheetFragment(
       .allowsDelayedPaymentMethods(allowsDelayedPaymentMethods ?: false)
       .defaultBillingDetails(defaultBillingDetails)
       .customer(
-        if (customerId.isNotEmpty() && customerEphemeralKeySecret.isNotEmpty()) PaymentSheet.CustomerConfiguration(
+        if (customerId.isNotEmpty() && customerSessionClientSecret.isNotEmpty()) PaymentSheet.CustomerConfiguration.createWithCustomerSession(
+          id = customerId,
+          clientSecret = customerSessionClientSecret
+        )
+        else if (customerId.isNotEmpty() && customerEphemeralKeySecret.isNotEmpty()) PaymentSheet.CustomerConfiguration(
```
Comment (charliecruzan-stripe) [inline android/src/main/java/com/reactnativestripesdk/PaymentSheetFragment.kt:200]: do we want to add error handling if they provide both `customerSessionClientSecret` and `customerEphemeralKeySecret`?

**Review by charliecruzan-stripe — CHANGES_REQUESTED** [review]
looking good, one nit about error handling and also could you add the type declaration for the new field to https://github.com/stripe/stripe-react-native/blob/master/src/types/PaymentSheet.ts#L13-L70 ?

File: android/src/main/java/com/reactnativestripesdk/PaymentSheetFragment.kt (hunk 1 above)
Comment (tjclawson-stripe) [reply android/src/main/java/com/reactnativestripesdk/PaymentSheetFragment.kt]: Updated 👍

**Comment by tjclawson-stripe** [comment]
> looking good, one nit about error handling and also could you add the type declaration for the new field to https://github.com/stripe/stripe-react-native/blob/master/src/types/PaymentSheet.ts#L13-L70 ?

Added the types to PaymentSheet.ts, but it doesn't seem to actually throw an error if both values are set. It's throwing an exception if both are set so it's fine, but is there a way for typescript to actually throw an error?

**Comment by charliecruzan-stripe** [comment]
> Added the types to PaymentSheet.ts, but it doesn't seem to actually throw an error if both values are set. It's throwing an exception if both are set so it's fine, but is there a way for typescript to actually throw an error?

No TS won't throw an error (depending on your typescript config, compilation _may_ fail) it'll just complain in the IDE, but that's good enough since we're also throwing an error in the native layer anyways

**Review by charliecruzan-stripe — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+***Features***
+- Adds support for CustomerSession in private beta [1744](https://github.com/stripe/stripe-react-native/pull/1744)
+
+***Fixes***
```
Comment (charliecruzan-stripe) [inline CHANGELOG.md:8]:
```suggestion
**Features**

- Adds support for CustomerSession in private beta [1744](https://github.com/stripe/stripe-react-native/pull/1744)

**Fixes**

```

File: android/src/main/java/com/reactnativestripesdk/PaymentSheetFragment.kt [hunk 3]
```diff
@@ -428,6 +428,31 @@ class PaymentSheetFragment(
...
+    private fun buildCustomerConfiguration(bundle: Bundle?): PaymentSheet.CustomerConfiguration? {
+      val customerId = bundle?.getString("customerId").orEmpty()
+      val customerEphemeralKeySecret = bundle?.getString("customerEphemeralKeySecret").orEmpty()
+      val customerSessionClientSecret = bundle?.getString("customerSessionClientSecret").orEmpty()
+      return if (customerSessionClientSecret.isNotEmpty() && customerEphemeralKeySecret.isNotEmpty()) {
+        throw PaymentSheetException("`customerEphemeralKeySecret` and `customerSessionClientSecret` cannot both be set")
+      } else if (customerId.isNotEmpty() && customerSessionClientSecret.isNotEmpty()) {
+        println("YEET creating customer configuration")
+        println("YEET customerID: $customerId")
+        println("YEET customerSessionClientSecret: $customerSessionClientSecret")
```
Comment (charliecruzan-stripe) [inline android/src/main/java/com/reactnativestripesdk/PaymentSheetFragment.kt:443]:
```suggestion
```

**Review by charliecruzan-stripe — CHANGES_REQUESTED** [review]

**Review by charliecruzan-stripe — APPROVED** [review]

**Pull request merged**